kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternDetailsController.java

```diff
@@ -255,6 +255,28 @@ private void initialize() {
         StringProperty fqnTextProperty = patternViewModel.getProperty(FQN_DESCRIPTION_NAME_TEXT);
         latestFqnText.textProperty().bind(fqnTextProperty);
 
+        // Update Other names section based on changes in List.
+        ObservableList<DescrName> descrNameObservableList = patternViewModel.getObservableList(OTHER_NAMES);
+        descrNameObservableList.addListener(new ListChangeListener<DescrName>() {
+            @Override
+            public void onChanged(Change<? extends DescrName> change) {
+                while(change.next()){
+                    if(change.wasAdded()){
+                        DescrName descrName = change.getAddedSubList().getFirst();
+                        List<TextFlow> rows = generateOtherNameRow(descrName);
+                        otherNamesVBox.getChildren().addAll(rows);
+                    } else if (change.wasRemoved()) {
+                        //clear all the vbox items/children since it is an overhead to track the
+                        //vBox child that has been modified.
+                        otherNamesVBox.getChildren().clear();
+                        //add the existing items back in vbox.
+                        List<TextFlow> rows = generateOtherNameRows();
+                        otherNamesVBox.getChildren().addAll(rows);
```

Comment:
instead of doing a clear first we could call setAll(..) here that replaces everything with what is passed in to setAll

Reply:
Thanks Pedro. Fixed it.

Reply:
Nice! I forgot to say that I think (pretty sure) that doing a setAll is slightly more performant than a clear and addAll